src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,36 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private MachineProducer<Bulldozer> bulldozerMachineProducer = new BulldozerProducer();
+    private MachineProducer<Excavator> excavatorMachineProducer = new ExcavatorProducer();
+    private MachineProducer<Truck> truckMachineProducer = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class type) {
+        return (type == Truck.class) ? new ArrayList<>(truckMachineProducer.get()) :
+                (type == Bulldozer.class) ? new ArrayList<>(bulldozerMachineProducer.get()) :
+                        (type == Excavator.class) ? new ArrayList<>(excavatorMachineProducer.get())
+                                : new ArrayList<>();
+    }
```

Comment:
It's a possible solution, but it's better to use if-else for improving readability